libs/api-implementation/orchestration/src/lib/orchestration.service.ts

```diff
@@ -15,26 +17,34 @@ export class VMOrchestrationService implements VMOrchestrationInterface {
     private zone;
     private machines: HashTable<VM>;
     private runningMachines: VM[];
-    private numOfTotalInstances;
+    private numOfTotalInstances: number;
 
     constructor() {
         this.compute = new Compute();
         this.zone = this.compute.zone('us-central1-c');
         this.numOfTotalInstances = 0;
-        this.runningMachines = []
+        this.runningMachines = [];
         this.machines = {};
     }
 
     async createMachine(name: string) {
         try {
-            const [vm, operation] = await this.zone.createVM(name, { os: 'ubuntu' });
-            this.machines[name] = new VM(name, Status.Running, vm);
+            const [vm, operation] = await this.createVM(name);
             this.runningMachines.push(this.machines[name]);
             this.numOfTotalInstances++;
-            console.log(operation)
+            console.log("Created the following machine: \n" + operation.metadata)
         } catch (error) {
-            console.error(error)
+            throwError(error);
+        }
+    }
+    
+    private async createVM(name) {
+        let vm, operation;
+        if (process.env.NODE_ENV != 'test') {
```

Comment:
This is not really a good idea to start making branches for tests. Preferably when running tests you mock the GCloud API so you are sure all logic is working as intended.